### PR TITLE
Add UnionField type

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

**Comment by vangheem** [comment]
@dmanchon elasticsearch allows single or array values by default. This is mostly to be more compatible with elasticsearch.

**Comment by dmanchon** [comment]
Yeah, that case its clearly fine, but do we want to limit somehow someone creating an indexed field of type: UnionType(Int, Date)? because that will be impossible to map on elasticsearch.

**Comment by vangheem** [comment]
well, for the UnionField implementation, it's fine; however, yes, we could add validation in guillotina_elasticsearch to handle this!

File: guillotina/api/search.py [hunk 1]
```diff
@@ -116,7 +117,7 @@ def __init__(self, context, request, security=False):
 
 
 @configure.service(
-    context=IResource, method='POST',
+    context=IContainer, method='POST',
```
Comment (bloodbare) [inline guillotina/api/search.py]: Why are we removing this feature, its good to be able to delete a catalog on a subresource

**Comment by vangheem** [comment]
@bloodbare 

> What is the use case of UnionField?

https://docs.python.org/3/library/typing.html#typing.Union

File: guillotina/api/search.py (hunk 1 above)
Comment (vangheem) [reply guillotina/api/search.py]: Do we? I don't understand why we would want this?

Comment (bloodbare) [reply guillotina/api/search.py]: Because the resource could be a subindex

File: guillotina/api/search.py [hunk 2]
```diff
@@ -3,12 +3,13 @@
...
 from guillotina.component import query_utility
 from guillotina.interfaces import ICatalogUtility
+from guillotina.interfaces import IContainer
 from guillotina.interfaces import IResource
 from guillotina.utils import get_content_path
 
 
 @configure.service(
-    context=IResource, method='GET', permission='guillotina.SearchContent', name='@search',
+    context=IContainer, method='GET', permission='guillotina.SearchContent', name='@search',
```
Comment (bloodbare) [inline guillotina/api/search.py]: Searching on a resource meaning that the resource is the basic path to search for.

File: guillotina/api/search.py (hunk 1 above)
Comment (vangheem) [reply guillotina/api/search.py]: alright, I guess

**Comment by vangheem** [comment]
@bloodbare 

> Shouldnt we have a way to define a field on allowed types on directives.py:117 ?

you can already customize the index definition. Anything with index field definition is beyond the scope of this PR though. I'm not touching that ehre.

**Comment by vangheem** [comment]
4.5.7 released

**Pull request merged**